File: src/menu.cpp
#include "menu.h"
#include "constants.h"

Menu::Menu(){
    std::cout << "\033[2J\033[1;1H"; //clean screen
    std::cout << "Neuronal Network C++.\n" << VERSION_NAME  << ". version: " << VERSION << "\n"<< std::endl;
    this->model = nullptr;
}

void Menu::set_model(Network * model){
    this->model = model;
}

Network * Menu::get_model(){
    return this->model;
}

void Menu::print_status(){
    std::cout << "Model name: " << (this->model != nullptr ? this->model->get_name() : "null" ) << "\n" << std::endl;
}

std::string Menu::input(){
    std::string input;
    std::cout << "<nnc "<<VERSION<<"> "<<VERSION_NAME<<"@"<<(this->model ? this->model->get_name() : "no-model")<<": ";
    getline(std::cin, input);
    return input;
}

File: src/menu.rs
pub struct Menu {
    model_loaded: i8,
    model_name: Option<String>,
    network: Option<String>, // TODO: change to Network
}

impl Menu {
    pub fn new() -> Menu {
        Menu {
            model_loaded: 0,
            model_name: None,
            network: None,
        }
    }

    pub fn print_status(&self) {
        println!("Model loaded: {}", self.model_loaded);
        println!("Model name: {}", self.model_name.as_ref().unwrap_or(&"null".to_string()));
    }

    pub fn input(&self) -> String {
        print!("<nnc {}> {}@{}: ", VERSION, VERSION_NAME, self.model_name.as_ref().unwrap_or(&"no-model".to_string()));
        let mut input = String::new();
        std::io::stdin().read_line(&mut input).unwrap();
        input
    }
}